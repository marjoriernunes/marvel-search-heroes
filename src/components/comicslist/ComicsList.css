h2 {
    color: #404040;
    margin-bottom: 5rem;
}

.container {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding-left: 0px;
    flex-direction: row;
}

.comic {
    list-style: none;
    align-items: flex-start;
    flex: 0 0 14%;
    margin: 0 0 3rem;
}

.comic img {
    background-size: cover;
}

.comic-text {
    width: 50%;
    color: #404040;
    font-weight: 700;
}

@media (max-width: 1050px) {
    .container {
        justify-content: center;
    }

    .comic {
        margin: 0 1rem;
    }
}