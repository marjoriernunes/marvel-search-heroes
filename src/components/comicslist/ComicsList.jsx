import './ComicsList.css';

const ComicsList = (props) => {
    return (
        <>
            <h2>Últimos lançamentos</h2>
            <ul className="container-comic">
                {props.data?.map((comic, index) => 
                   ( 
                        <li key={index} className="comic">
                            <img 
                                alt={comic.title} 
                                src={`${comic.thumbnail.path}/portrait_xlarge.${comic.thumbnail.extension}`} />
                            <p className="comic-text">{comic.title}</p>
                        </li>   
                    )
                )}
            </ul>
        </>
    );
}

export default ComicsList;