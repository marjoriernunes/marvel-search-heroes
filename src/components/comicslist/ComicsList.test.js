import { render } from '@testing-library/react';
import ComicsList from './ComicsList';

test('renders correctly', () =>{
    const utils = render(<ComicsList/>)
    expect(utils.container).toMatchSnapshot()
});