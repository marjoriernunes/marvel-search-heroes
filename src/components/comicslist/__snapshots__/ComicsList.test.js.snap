// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Últimos lançamentos
  </h2>
  <ul
    class="container"
  />
</div>
`;
